experimental/stats/metricregistry_test.go

```diff
@@ -0,0 +1,260 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package stats
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/internal/grpctest"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+// TestPanic tests that registering two metrics with the same name across any
+// type of metric triggers a panic.
+func (s) TestPanic(t *testing.T) {
+	snapshotMetricsRegistryForTesting(t)
+	want := "metric simple counter already registered"
+	defer func() {
+		if r := recover(); !strings.Contains(r.(string), want) {
+			t.Errorf("expected panic contains %q, got %q", want, r)
+		}
+	}()
+	desc := MetricDescriptor{
+		// Type is not expected to be set from the registerer, but meant to be
+		// set by the metric registry.
+		Name:        "simple counter",
+		Description: "number of times recorded on tests",
+		Unit:        "calls",
+	}
+	RegisterInt64Count(desc)
+	RegisterInt64Gauge(desc)
+}
```

Comment:
This test doesn't fail if a panic doesn't occur.

Reply:
recover() returns nil if the test is not panicing. Then I typecast to string, and that won't contain the want right? Will my typecast of a nil to a string error?

Reply:
Oh, duh, the defer will still run, sorry!